pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -149,3 +149,26 @@ func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
 
 	return copy, nil
 }
+
+func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.GracefulDeleter)
+	if !ok {
+		return nil, false, errDualWriterDeleterMissing
+	}
+
+	deletedLS, async, err := legacy.Delete(ctx, name, deleteValidation, options)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.FromContext(ctx).Error(err, "could not delete from legacy store", "mode", 2)
```

Comment:
does klog give us the name/key being deleted? if not we should add it so we can tell which item the error refers to

Reply:
There is a `WithValues` method we could use, I guess? I guess I could log the resource version in here.

Reply:
but I'll address better logging in a different PR if that's alright :)

Reply:
For now we could add it the same way we have in some of the other logs:

```
klog.FromContext(ctx).Error(err, "could not delete from legacy store", "mode", 2, "name", name)
```